### ISSUE TITLE
Y-Axis Crash Detection Inconsistancies

### ISSUE THREAD

**Issue by TheZeroBeast** [prusa3d/Prusa-i3-Plus#76]
#### Printer Setup
 * MK3 on 3.2.1 FW
 * Running latest R2 Printed Parts
 * Slic3r 1.40.0-Beta

#### The Issue
I have had continuous issues with Y-Axis on crash detection being offset by different amounts each crash.
This results in a small layer shift in print rendering the print a failure.

As the **Y-Axis** is using tmc2130 sensorless homing it relies on hitting the first bearing which is an offset load causing a random amount of skew from bearings and the offset load.

#### The Solution
I have redesigned the **Y-Motor-holder** to incorporate a hard stop for the hotbed belt mount so the load is centered to the motor and removes the random skew introduced.   _**PR #74 #77**_

#### The Testing
This has been tested on my MK3 with single wall cubes with flawless results as per attached photo. Extrusion issues seen are at crash/recovery point which could be improved in FW side.
![img_2962](https://user-images.githubusercontent.com/20939095/40747302-6399863c-644c-11e8-9909-baa9904368d4.jpg)
![img_2963](https://user-images.githubusercontent.com/20939095/40747301-632cd0a0-644c-11e8-8372-0958ced89e52.jpg)

**Comment by 3d-gussner** [comment]
@TheZeroBeast Nicely done! 
As the ball bearings are not really fixed by the U-bolts and can move a bit when hit the y-rod-holder it makes sense to use the reliable MK2/s design where the y-belt-holder hits the y-motor-holder.

BTW: The STL seams to be broken.

**Comment by hst68** [comment]
can confirm it: The STL seams to be broken.

**Comment by TheZeroBeast** [comment]
Ok @hst68 @3d-gussner all good now, thank you for the heads up. 👍😂
Rob

**Comment by 3d-gussner** [comment]
STL is fixed

**Comment by hst68** [comment]
now I can't find the file....

**Comment by jltx1** [comment]
Why not extend a tab off the new extension to add a third screw hole to tighten up the Y motor?  It's a touch flexy as is.  Here is a concept.  Need to thin the tab some so screw clears the Y belt holder.  But you get the idea.

![untitled](https://user-images.githubusercontent.com/28034983/40870431-bfd192c4-65f4-11e8-8477-3bf75a3d520a.png)

**Comment by 3d-gussner** [comment]
@jltx1 Nice idea,  as you said it has to be bit thinner as there is around 6mm space between motor and y-belt holder.

**Comment by jltx1** [comment]
Yes, I thinned it out which means it needs M3x6mm screw.  But I tested it and it works great.  Thanks, TheZeroBeast!

**Comment by gregsaun** [comment]
Thank you for this finding and for sharing it!

**Comment by 3d-gussner** [comment]
@jltx1 As the third screw plastic just needs to be 1mm thinner i think that the M3x10 should fit.

**Comment by jltx1** [comment]
Nope.  Tried it.  Motor bore is too shallow.  Would block the Y carriage.  8mm *almost* makes it.  6mm works great.  Sad I know.

**Comment by gregsaun** [comment]
@TheZeroBeast I have done some measurements and you are using a length of 28.1mm from top motor hole to end of your added end stop. This is a good value if bearing is almost centered. However, if bearing is not centered and pushed in direction of the Y motor, length might become more 28.9-29mm.

**Comment by jltx1** [comment]
I noticed the same.  I decided to leave it so as not to reduce Y range.  IMO, better to update assembly instructions to have folks check that the center hits first and adjust bearing as needed. 

-- Jon

**Comment by TheZeroBeast** [comment]
Hi @gregsaun, I designed this to limit the loss of y-axis range and agree with @jltx1 that if incorporated build instructions should be updating to accomadate.

On a side note I have done half a dozon 6h test prints to heights over 85mm without and Y-Axis crashes detected (confirmed it is still turned on)
Previously Y-Axis crashes would happen all the time for apparently no reason and with the feature off no steps are lost. Last time I looked my all time crash count was over 400 for Y and under 30 for X.

I feel that as we retuned the belt tention/crash threshold after the installation of this part it has come to a more accurate value resulting in more robust functioning.

I'd be interested to hear how you have found your prints since this upgrade @jltx1

**Comment by alientek** [comment]
where can i download this?

**Comment by jltx1** [comment]
There are at least two versions on thingiverse.  Here’s my remix of the original https://www.thingiverse.com/thing:2942590

-- Jon

**Comment by VVlasy** [comment]
This has died down but I still feel this is an issue.

I get at least one to two crashes on Y axis when printing a benchy.

I rebuilt and readjusted tension on both the Y axis and X axis.
I also carefully tightened the u bolts just to the point when they are not loose. I cannot feel any resistance moving either axis back and forth. even with the belts both on and off. 

Currently the values for belt state is X 283 and Y 266.

i3 MK3S kit, FW 3.8.1

**Comment by catid** [comment]
In my case, using the Prusa V2 enclosure to print ASA, I was seeing frequent Y crashes above a certain Z level. The issue was caused by having the printer too far back into the enclosure, so that the cable bundle coming off the bed was hitting the back window. This also caused the belt self test results to report unexpectedly high values (270) or failing. By moving the printer forward to avoid this, the belt self test result is now reading 240. Furthermore this fixed the issue where crash detection was triggering during printing.

**Comment by Prusa-Support** [comment]
Thank you for sharing your idea.

The modification suggested in this issue could reduce the bounce across the right side of the Y-carriage: Y-axis homes once the rear bearing on the left side hits the end of the axis, and the one-bearing on the right side may have slight backlash instead.
I personally like the suggested design improvement, nonetheless, such bounce may be negligible or not concerning. In fact, the "end-stop linear-bearing" will remain a firm point of reference for the Y-homing.

Moreover, I'd like to highlight that in case of numerous crash detections, the main concern should move to that aspect in the first place, and possibly be sorted out with proper troubleshooting.

This doesn't mean that our developers can't work for a more reliable Y-homing 🙂 but, so far, the "bounce" is not a confirmed issue and this issue could be closed as stale, but please share your thoughts and let us know if the problem persists.

Please consider contacting the Technical Support before posting about new similar problems.
https://help.prusa3d.com/article/customer-support_2287

Michele Moramarco
Prusa Research
